src/main/native/smithwaterman/PairWiseSW.h

```diff
@@ -410,20 +409,35 @@ void inline getCIGAR(SeqPair *p, int16_t *cigarBuf_, int32_t tid)
                 state = 'R';
                 break;
         }
-	// expectedLength for converting int to str w/ extra padding for '\0'
-	int expectedLength = snprintf( NULL, 0, "%d%c", cigarArray[2 * i + 1], state);
-	if (curSize >= 0 && expectedLength > 1 && (curSize < maxSize - expectedLength ))
-           {
-                   curSize += snprintf(p->cigar + curSize, expectedLength + 1, "%d%c", cigarArray[2 * i + 1], state);
-           }
+        // expectedLength for converting int to str w/ extra padding for '\0'
+	    int expectedLength = snprintf( NULL, 0, "%d%c", cigarArray[2 * i + 1], state);
+        if (curSize >= 0 && expectedLength > 1){ 
```

Comment:
Please check if PR #148 addresses changes are compatible. If so, we can simply replace maxSize in previous condition w/ cigarBufLen. I like you brought cigarBufLength; and there is no API change.